### PR TITLE
[notie] Update notie types for v4.3

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Related discussion: #59932

Types for the `notie` package were very out of date (types were for 0.0 and current version is 4.3), so these types were written from scratch.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test notie`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: All methods are documented in the README
  - <https://github.com/jaredreich/notie#available-methods>
  - <https://github.com/jaredreich/notie#options--methods>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by MysteryBlokHed** [comment]
@peterblazejewicz — Would you or another maintainer mind taking a look at this if you have a chance? I'm not sure if the original author is still active, and this PR basically adds new definitions because of how big it is.

File: types/notie/index.d.ts [hunk 1]
```diff
@@ -1,10 +1,192 @@
-// Type definitions for notie.js
-// Project: https://github.com/jaredreich/notie.js
-// Definitions by: Mateus Demboski <https://github.com/mateusdemboski>
+// Type definitions for notie 4.3
+// Project: https://github.com/jaredreich/notie
+// Definitions by: Adam Thompson-Sharpe <https://github.com/MysteryBlokHed>
+//                 Mateus Demboski <https://github.com/mateusdemboski>
```
Comment (peterblazejewicz) [inline types/notie/index.d.ts:4]:
```suggestion
// Definitions by: Mateus Demboski <https://github.com/mateusdemboski>
//                 Adam Thompson-Sharpe <https://github.com/MysteryBlokHed>
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@MysteryBlokHed thx!

**Comment by MysteryBlokHed** [comment]
Ready to merge

**Pull request merged**